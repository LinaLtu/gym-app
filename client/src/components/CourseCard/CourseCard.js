import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import './CourseCard.scss';

const CSS_NAME = 'course-card';

const CourseCard = ({ course }) => {
    const history = useHistory();

    const handleOnClick = () => {
        history.push(`/course/${course._id}`)
    }

    return (
        <div className={CSS_NAME} onClick={handleOnClick}>
            <div className={`${CSS_NAME}_course-name`}>{course.name}</div>
            <div className={`${CSS_NAME}_details`}>{course.startDate} | {course.category}</div>
            <div className={`${CSS_NAME}_description`}> {course.description}</div>
        </div >
    )
}

CourseCard.propTypes = {
    course: PropTypes.object.isRequired
};


export default CourseCard;