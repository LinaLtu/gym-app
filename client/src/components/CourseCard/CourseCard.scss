@import '../shared/common.scss';

.course-card {
    border: 1px solid white;
    border-radius: 5px;
    font-weight: bold;
    width: 255px;
    height: 200px;
    margin: 20px;
    color: white;
    background-color: $brand_color;
    box-shadow: 0 0 6px 0 rgba(0,0,0,0.2);

    &_course-name {
        font-size: 32px;
        padding: 20px 10px;
    }

    &_details {
        padding: 10px;
    }

    &_description {
        padding: 10px;
        visibility: hidden;
    }

}

.course-card:hover {
    cursor: pointer;
    opacity: 0.7;

    .course-card_description {
        visibility: visible;
        transition: all 2s ease-in-out;
    }  
}
