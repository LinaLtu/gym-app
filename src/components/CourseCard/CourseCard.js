import React, { useState, Fragment } from 'react';
import './CourseCard.scss';

const CSS_NAME = 'course-card';

const mockCourseList = [
    {
        name: 'zumba',
        description: 'latin music and good mood guaranteed',
        startDate: 'monday',
        category: 'c2'
    },
    {
        name: 'body attack',
        description: 'intensive and sweaty',
        startDate: 'tuesday',
        category: 'c3'
    },
    {
        name: 'body balance',
        description: 'low impact',
        startDate: 'friday',
        category: 'c1'
    },
]

const CourseCard = () => {

    const [courseList, setCourseList] = useState(mockCourseList);

    const courseCardTemplate = courseList.map(course => {
        return (
            <div className={CSS_NAME}>
                <div>Course Name: {course.name}</div>
                <div>Description: {course.description}</div>
                <div>Start Date: {course.startDate}</div>
                <div>Category: {course.category}</div>
            </div>
        )
    });

    return (
        courseCardTemplate
    )
}

export default CourseCard;