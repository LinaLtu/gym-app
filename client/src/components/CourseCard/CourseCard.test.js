import React from 'react';
import renderer from 'react-test-renderer';
import reactRouterDom from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import CourseCard from '.';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({ useHistory: jest.fn() }));

const mockCourse = {
    _id: 'id',
    name: 'name',
    startDate: '',
    categories: ['c1', 'c2']
};

describe('CourseCard Component', () => {
    const pushMock = jest.fn();

    beforeEach(() => {
        reactRouterDom.useHistory.mockImplementation(() => ({
            push: pushMock
        }));
    });

    it('renders according to design', () => {
        const component = renderer.create(
            <CourseCard course={mockCourse} />
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    describe('#handleOnClick', () => {
        let wrapper

        beforeEach(() => {
            wrapper = shallow(<CourseCard course={mockCourse} />);
        });

        it('#setCategories callback gets called', () => {
            wrapper.find('.course-card').simulate('click');
            expect(pushMock).toHaveBeenCalled();
        });
    });
});