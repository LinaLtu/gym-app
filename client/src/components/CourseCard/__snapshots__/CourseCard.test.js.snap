// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard Component renders according to design 1`] = `
<div
  className="course-card"
  onClick={[Function]}
>
  <div
    className="course-card_course-name"
  >
    name
  </div>
  <div
    className="course-card_details"
  >
    
     | 
  </div>
  <div
    className="course-card_description"
  >
     
  </div>
</div>
`;
